validation/citrus-validation-json/src/main/java/org/citrusframework/validation/json/JsonTextMessageValidator.java

```diff
@@ -56,225 +46,85 @@
-            logger.debug("Skip message payload validation as no control message was defined");
-            return;
-        }
-
+    public void validateMessage(
+            Message receivedMessage,
+            Message controlMessage,
+            TestContext context,
+            JsonMessageValidationContext validationContext
+    ) {
         logger.debug("Start JSON message validation ...");
-
         if (validationContext.isSchemaValidationEnabled()) {
             jsonSchemaValidation.validate(receivedMessage, context, validationContext);
         }
 
         String receivedJsonText = receivedMessage.getPayload(String.class);
         String controlJsonText = context.replaceDynamicContentInString(controlMessage.getPayload(String.class));
 
-        try {
-            if (!StringUtils.hasText(controlJsonText)) {
-                logger.debug("Skip message payload validation as no control message was defined");
-                return;
-            } else {
-                if (!StringUtils.hasText(receivedJsonText)) {
-                    throw new ValidationException("Validation failed - " +
-                		"expected message contents, but received empty message!");
-                }
-            }
-
-            JSONParser parser = new JSONParser(permissiveMode);
-
-            Object receivedJson = parser.parse(receivedJsonText);
-            ReadContext readContext = JsonPath.parse(receivedJson);
-            Object controlJson = parser.parse(controlJsonText);
-            if (receivedJson instanceof JSONObject) {
-                validateJson("$.", (JSONObject) receivedJson, (JSONObject) controlJson, validationContext, context, readContext);
-            } else if (receivedJson instanceof JSONArray) {
-                JSONObject tempReceived = new JSONObject();
-                tempReceived.put("array", receivedJson);
-                JSONObject tempControl = new JSONObject();
-                tempControl.put("array", controlJson);
-
-                validateJson("$.", tempReceived, tempControl, validationContext, context, readContext);
-            } else {
-                throw new CitrusRuntimeException("Unsupported json type " + receivedJson.getClass());
-            }
-        } catch (IllegalArgumentException e) {
-            throw new ValidationException(String.format("Failed to validate JSON text:%n%s", receivedJsonText), e);
-        } catch (ParseException e) {
-            throw new CitrusRuntimeException("Failed to parse JSON text", e);
+        if (!hasText(controlJsonText)) {
+            logger.debug("Skip message payload validation as no control message was defined");
+            return;
+        } else if (!hasText(receivedJsonText)) {
+            throw new ValidationException("Validation failed - expected message contents, but received empty message!");
         }
 
+        new JsonElementValidator(strict, context, validationContext.getIgnoreExpressions()).validate(
```

Comment:
How about sharing the instance of JsonElementValidator in this class as a private member. Users may want to set a customized implementation of JsonElementValidator then.

We can pass the parameters to the JsonElementValidator#validate method then (e.g. context and validationContext) so JsonElementValidator is stateless

Reply:
I've introduced a `JsonElementValidator.Provider` to configure `JsonTextMessageValidator` with a custom element validator - hope that's fine?